### ISSUE TITLE
incorrect GPU information

### ISSUE THREAD

**Issue by Mr-nUUb** [flightlessmango/MangoHud#725]
MangoHud is displaying incorrect GPU information, at least clocks and power consumption on AMD systems.
Please see the attached video which shows MangoHud and CPU-X.

System information:
* Manjaro Linux, up 2 date
* Kernel 5.16.14-1-MANJARO
* Mesa 21.3.7-2
* MangoHud 0.6.6.1.r181.g284ae0d-1
* KDE Plasma 5.23.4
* Wayland session
* CPU: AMD Ryzen 7 5800X
* GPU: AMD Radeon RX 6900 XT
* RAM: 32 GB DDR4-3600

https://user-images.githubusercontent.com/36689675/162048546-f6845a17-e88b-46a9-b03d-4d096ec911b1.mp4

**Comment by jackun** [comment]
Weird, it's exactly the same methods. Try with both using 1sec sampling period: `fps_sampling_period=1000`

**Comment by Mr-nUUb** [comment]
With `fps_sampling_period=1000` it's... the same, but at greater intervals.
* In menu screens it shows as low as 4 MHz without ever going above 109 MHz.
* During actual gameplay it's between 17 and 2660 MHz, completely unstable.
It seems like that firmware misbehavior mentioned in other issues. I think the only solution to this is reading metrics multiple times and building a mean value.

**Comment by flightlessmango** [comment]
gpu load should be more representative in latest, can you check? 5349226fa50f98c7d3328258112f48865b96cddb

**Comment by Mr-nUUb** [comment]
I'm on e4b23c9 and load is looking good now. AFAIK this commit changes the logic behind load only, right?

**Comment by Mr-nUUb** [comment]
@flightlessmango I built https://github.com/flightlessmango/MangoHud/commit/ae85730448f3ac7c895e5669f48aab032abb3040 yesterday, and it's looking great. However, the core clock is below 50 MHz most of the time, while load is above 70%. I think using the last value isn't appropriate and an average should be used instead. Maybe this also applies to memory clock, but I didn't check.

**Comment by flightlessmango** [comment]
changed it to average, does it look good in the hud? ec5952d834ca3ab7fc6f4f9028663d1ac2c8efce

**Comment by Mr-nUUb** [comment]
It's better, but something is still off.

https://user-images.githubusercontent.com/36689675/166971313-4e10767c-246d-4b1e-be89-6923d390fce7.mp4

**Comment by flightlessmango** [comment]
What am I looking for exactly?

**Comment by Mr-nUUb** [comment]
Why is there a difference of ~500 MHz between mangohud and radeontop?

**Comment by flightlessmango** [comment]
They might be showing the max value for the period instead of average

**Comment by flightlessmango** [comment]
Does this patch make it inline with radeontop?
```diff --git a/src/amdgpu.cpp b/src/amdgpu.cpp
index a147ffe..96ebc76 100644
--- a/src/amdgpu.cpp
+++ b/src/amdgpu.cpp
@@ -165,8 +165,8 @@ void amdgpu_metrics_polling_thread() {
 		UPDATE_METRIC_AVERAGE(average_gfx_power_w);
 		UPDATE_METRIC_AVERAGE(average_cpu_power_w);
 
-		UPDATE_METRIC_AVERAGE(current_gfxclk_mhz);
-		UPDATE_METRIC_AVERAGE(current_uclk_mhz);
+		UPDATE_METRIC_MAX(current_gfxclk_mhz);
+		UPDATE_METRIC_MAX(current_uclk_mhz);
 
 		UPDATE_METRIC_MAX(soc_temp_c);
 		UPDATE_METRIC_MAX(gpu_temp_c);
```

**Comment by Mr-nUUb** [comment]
I don't get it, now mangohud is ~1 GHz higher than radeontop, but at least it doesn't drop down to 4 MHz anymore.

**Comment by Mr-nUUb** [comment]
Fixed by 08431a0221623fa83e3c884ccc0b03e53f7a7eda
